ros/src/system/autoware_health_checker/src/health_aggregator.cpp

```diff
@@ -0,0 +1,114 @@
+#include <autoware_health_checker/health_aggregator.h>
+
+HealthAggregator::HealthAggregator(ros::NodeHandle nh,ros::NodeHandle pnh)
+{
+    nh_ = nh;
+    pnh_ = pnh;
+}
+
+HealthAggregator::~HealthAggregator()
+{
+
+}
+
+void HealthAggregator::run()
+{
+    system_status_pub_ = nh_.advertise<autoware_system_msgs::SystemStatus>("/system_status",10);
+    node_status_sub_ = nh_.subscribe("/node_status",10,&HealthAggregator::nodeStatusCallback,this);
+    diagnostic_array_sub_ = nh_.subscribe("/diagnostic_agg",10,&HealthAggregator::diagnosticArrayCallback,this);
+    boost::thread publish_thread(boost::bind(&HealthAggregator::publishSystemStatus, this));
+    return;
+}
+
+void HealthAggregator::publishSystemStatus()
+{
+    ros::Rate rate = ros::Rate(autoware_health_checker::UPDATE_RATE);
+    while(ros::ok())
+    {
+        mtx_.lock();
+        system_status_.header.stamp = ros::Time::now();
+        updateConnectionStatus();
+        system_status_pub_.publish(system_status_);
+        system_status_.node_status.clear();
+        system_status_.hardware_status.clear();
+        mtx_.unlock();
+        rate.sleep();
+    }
+    return;
+}
+
+void HealthAggregator::updateConnectionStatus()
+{
+    std::vector<std::string> detected_nodes;
+    ros::master::getNodes(detected_nodes);
+    system_status_.available_nodes = detected_nodes;
+    return;
+}
+
+void HealthAggregator::nodeStatusCallback(const autoware_system_msgs::NodeStatus::ConstPtr msg)
+{
+    mtx_.lock();
+    system_status_.node_status.push_back(*msg);
+    mtx_.unlock();
+    return;
+}
+
+void HealthAggregator::diagnosticArrayCallback(const diagnostic_msgs::DiagnosticArray::ConstPtr msg)
+{
+    mtx_.lock();
+    boost::optional<autoware_system_msgs::HardwareStatus> status = convert(msg);
```

Comment:
The contents of `convert()` are closed and I think that it can be removed from mutex.